apps/email/js/cards/base_card.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+define(function(require) {
+  var mozL10n = require('l10n!'),
+      base = require('./base');
+
+  // Set up the global time updates for all nodes.
+  (function() {
+    var formatter = new mozL10n.DateTimeFormat();
+    var updatePrettyDate = function updatePrettyDate() {
+      var labels = document.querySelectorAll('[data-time]');
+      var i = labels.length;
+      while (i--) {
+        labels[i].textContent = formatter.fromNow(
+          labels[i].dataset.time,
+          // the presence of the attribute is our indicator; not its value
+          'compactFormat' in labels[i].dataset);
+      }
+    };
+    var timer = setInterval(updatePrettyDate, 60 * 1000);
+
+    function updatePrettyDateOnEvent() {
+      clearTimeout(timer);
+      updatePrettyDate();
+      timer = setInterval(updatePrettyDate, 60 * 1000);
+    }
+    // When user changes the language, update timestamps.
+    mozL10n.ready(updatePrettyDateOnEvent);
+
+    // On visibility change to not hidden, update timestamps
+    document.addEventListener('visibilitychange', function() {
+      if (document && !document.hidden) {
+        updatePrettyDateOnEvent();
+      }
```

Comment:
(I know this is just moved code, but): Maybe we should call clearInterval in an else case here too so we can be extra good CPU citizens?